pkg/controller/importconfig/render.go

```diff
@@ -31,6 +37,137 @@ type ImagePullSecretConfig struct {
 	ImagePullSecretType      corev1.SecretType
 }
 
-const (
-	AgentRegistrationDefaultBootstrapSAName = "agent-registration-bootstrap"
-)
+type KlusterletManifestsConfig struct {
+	Files       []string
+	InstallMode operatorv1.InstallMode
+
+	ClusterName                  string
+	KlusterletNamespace          string
+	KlusterletClusterAnnotations map[string]string
+	BootstrapKubeconfig          []byte
+
+	// In the managed cluster annotations, it contains nodeSelectors and tolerations for the klusterlet deployment.
+	ManagedClusterAnnotations map[string]string
+
+	generateImagePullSecret bool // by default is true, in hosted mode, it will be set false
+}
+
+func NewKlusterletManifestsConfig(files []string, installMode operatorv1.InstallMode,
+	clusterName, klusterletNamespace string, bootstrapKubeconfig []byte) *KlusterletManifestsConfig {
+	return &KlusterletManifestsConfig{
+		Files:                   files,
+		InstallMode:             installMode,
+		ClusterName:             clusterName,
+		KlusterletNamespace:     klusterletNamespace,
+		BootstrapKubeconfig:     bootstrapKubeconfig,
+		generateImagePullSecret: true,
+	}
+}
+
+func (c *KlusterletManifestsConfig) WithKlusterletClusterAnnotations(a map[string]string) *KlusterletManifestsConfig {
+	c.KlusterletClusterAnnotations = a
+	return c
+}
+
+func (c *KlusterletManifestsConfig) WithManagedClusterAnnotations(a map[string]string) *KlusterletManifestsConfig {
```

Comment:
comment on what resource's annotation you are looking at.

Reply:
Comment added.